### PR TITLE
Speed up decompressing large cab archives by 30%

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
@superm1 does this make any more sense to you than me? It works find locally for me on fedora 41, but not on CI. Ubuntu, but not Debian, etc.

**Comment by superm1** [comment]
> @superm1 does this make any more sense to you than me? It works find locally for me on fedora 41, but not on CI. Ubuntu, but not Debian, etc.

just pushed what stands out to me

File: libfwupdplugin/fu-self-test.c [hunk 1]
```diff
@@ -5742,6 +5743,34 @@ fu_lzma_func(void)
 	g_assert_true(ret);
 }
 
+static void
+fu_cab_checksum_func(void)
+{
+	guint8 buf[] = {0x10, 0x20, 0x30, 0x40, 0x50, 0x60, 0x70, 80};
```
Comment (superm1) [inline libfwupdplugin/fu-self-test.c:5749]: is that last one supposed to be _0x80_?

Comment (hughsie) [reply libfwupdplugin/fu-self-test.c]: Ohh yes, fixed -- but that was just test data I was comparing against the old implementation and the new implementation. I've tested it with asan, valgrind, this local test data and in the end-to-end tests and I can't work out what the problem is with the new code...

**Comment by superm1** [comment]
Is this possibly a case of parallel tests in CI?

**Comment by hughsie** [comment]
> Is this possibly a case of parallel tests in CI?

At this point I'm considering _aliens_. I'm splitting up each micro optimisation in https://github.com/fwupd/fwupd/pull/8142 and I'll wait for the CI to fail after some commit. Patience I suppose. :)

**Comment by superm1** [comment]
Well try turning off parallel tests:

https://mesonbuild.com/Unit-tests.html#parallelism

That will rule out the race

**Review by superm1 — APPROVED** [review]

**Pull request merged**